core/Db/TransactionLevel.php

```diff
@@ -40,12 +40,17 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        if ($this->db->supportsUncommitted !== null) {
+            return $this->db->supportsUncommitted;
+        }
```

Comment:
We should only skip the rest if `supportsUncommitted` was set to false.

```suggestion
        if ($this->db->supportsUncommitted === false) {
            return false;
        }
```

Reply:
👍  absolutely, otherwise we never set the uncommitted status again when it is supported. @peterhashair I had also sent you this in slack